File: Command.java
/*****
 * Command
 * Encapsulate a request as an object,
 * thereby letting you parameterize clients with different requests,
 * queue or log requests, and support undoable operations.
*****/

import java.util.List;
import java.util.ArrayList;


interface Command_ { void execute(); }

class Receiver { public void method() { System.out.println("Receiver"); } }

class ConcreteCommand implements Command_ {
    private Receiver receiver;

    public ConcreteCommand(Receiver receiver) { this.receiver = receiver; }

    @Override
    public void execute() { receiver.method(); }
}

class Invoker {
    private List<Command_> commands = new ArrayList<>();

    public void add(Command_ command) { commands.add(command); }
    public void execute() { for (Command_ command : commands)  command.execute(); }
}


public class Command {
    public static void main(String[] args) {
        Receiver r = new Receiver();
        Invoker i = new Invoker();

        Command_ cmd = new ConcreteCommand(r);
        i.add(cmd);
        i.execute();
    }
}

File: Command.js
/*****
 * Command
 * Encapsulate a request as an object,
 * thereby letting you parameterize clients with different requests,
 * queue or log requests, and support undoable operations.
*****/


class Command { execute() {} }

class Receiver { method() { console.log("Receiver"); } }

class ConcreteCommand extends Command {
    constructor(receiver) {
        super();
        this.receiver = receiver;
    }

    execute() { this.receiver.method(); }
}

class Invoker {
    constructor() { this.commands = []; }

    add(command) { this.commands.push(command); }
    execute() { for (const command of this.commands) command.execute(); }
}


const r = new Receiver();
const i = new Invoker();

const cmd = new ConcreteCommand(r);
i.add(cmd);
i.execute();
